src/benchmarklib/benchmark_table_encoder.cpp

```diff
@@ -98,19 +97,13 @@ bool BenchmarkTableEncoder::encode(const std::string& table_name, const std::sha
     }
 
     // No column-specific or type-specific encoding was specified.
-    // Use default if it is compatible with the column type or leave column Unencoded if it is not.
-    if (encoding_supports_data_type(encoding_config.default_encoding_spec.encoding_type, column_data_type)) {
-      chunk_encoding_spec.push_back(encoding_config.default_encoding_spec);
-    } else {
-      // Use a stringstream here to bundle all writes into a single one and avoid locking.
-      auto output = std::ostringstream{};
-      output << " - Column '" << table_name << "." << table->column_name(column_id) << "' of type ";
-      output << column_data_type << " cannot be encoded as ";
-      output << encoding_config.default_encoding_spec.encoding_type << " and is ";
-      output << "left Unencoded.\n";
-      std::cout << output.str();
-      chunk_encoding_spec.emplace_back(EncodingType::Unencoded);
+    if (default_encoding_spec && encoding_supports_data_type(default_encoding_spec->encoding_type, column_data_type)) {
+      chunk_encoding_spec.push_back(*default_encoding_spec);
+      continue;
     }
+
+    const auto column_is_unique = table->column_is_unique(column_id);
+    chunk_encoding_spec.push_back(auto_select_segment_encoding_spec(column_data_type, column_is_unique));
```

Comment:
Why no longer print?

Reply:
I felt like this print was meant as a warning that a column is left unencoded. But I guess I could issue a warning in case the default encoding is ignored because it is not supported.

Reply:
Yes, it is still a helpful message imo.